test/unified_memory_allocation/memoryPool.hpp

```diff
@@ -73,6 +76,56 @@ TEST_P(umaPoolTest, pow2AlignedAlloc) {
     }
 }
 
+TEST_P(umaPoolTest, allocZeroSize) {
+    static constexpr size_t allocSize = 0;
+    auto *ptr = umaPoolMalloc(pool.get(), allocSize);
```

Comment:
Should ptr be a nullptr or a valid ptr in this case? Please, add an assert accordingly.